app/src/main/java/com/example/habitsmasher/ui/dashboard/AddHabitDialog.java

```diff
@@ -0,0 +1,186 @@
+import com.google.firebase.firestore.EventListener;
+import com.google.firebase.firestore.FirebaseFirestore;
+import com.google.firebase.firestore.FirebaseFirestoreException;
+import com.google.firebase.firestore.QueryDocumentSnapshot;
+import com.google.firebase.firestore.QuerySnapshot;
+
+import java.text.ParseException;
+import java.text.SimpleDateFormat;
+import java.util.ArrayList;
+import java.util.Date;
+import java.util.HashMap;
+
+/**
+ * Making a dialogfragment from fragment came from the following video:
+ * https://www.youtube.com/watch?v=LUV_djRHSEY
+ *
+ * Name: Mitch Tabian
+ * Video Date: December 10, 2017
+ */
+public class AddHabitDialog extends DialogFragment{
+
+    private static final String TAG = "AddHabitDialog";
+    FirebaseFirestore db;
+
+    public  interface HabitDialogListener {
+        void addNewHabit(String title, String reason, Date date);
+    }
+
+    public HabitDialogListener _listener;
+
+    /**
+     * Initializing EditTexts and dialog listener.
+     */
+    private EditText habitTitleEditText;
+    private EditText habitReasonEditText;
+    private EditText habitDateEditText;
+    private Button confirmNewHabit;
+    private Button cancelNewHabit;
+
+
+    @Nullable
+    @Override
+    public View onCreateView(@NonNull LayoutInflater inflater, @Nullable ViewGroup container, @Nullable Bundle savedInstanceState) {
+        View view = inflater.inflate(R.layout.add_habit_dialog_box, container, false);
+        habitTitleEditText = view.findViewById(R.id.habit_title_edittext);
+        habitReasonEditText = view.findViewById(R.id.habit_reason_edittext);
+        habitDateEditText = view.findViewById(R.id.habit_date_edittext);
+        confirmNewHabit = view.findViewById(R.id.confirmHabit);
+        cancelNewHabit = view.findViewById(R.id.cancelHabit);
+
+        db = FirebaseFirestore.getInstance();
+        /**
+         * Name of database collection.
+         */
+        final CollectionReference collectionReference = db.collection("Habits");
+
+        /**
+         * Closes Dialog box if user presses the cancel button.
+         */
+        cancelNewHabit.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View view) {
+                Log.d(TAG, "Cancel");
+                getDialog().dismiss();
+            }
+        });
+
+        confirmNewHabit.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View view) {
+                Log.d(TAG, "Confirm");
+                /**
+                 * Gets user input and saves to string variables.
+                 */
+                String habitTitleInput = habitTitleEditText.getText().toString();
+                String habitReasonInput = habitReasonEditText.getText().toString();
+                String habitDateInput = habitDateEditText.getText().toString();
+                SimpleDateFormat inputDateFormatter = new SimpleDateFormat("dd-MM-yyyy");
```

Comment:
We can probably extract the date pattern string on this line to it's own **constant**